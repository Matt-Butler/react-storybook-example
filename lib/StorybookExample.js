'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

require('../src/StorybookExample.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var StorybookExample = function StorybookExample(props) {
  return _react2.default.createElement(
    'h1',
    { className: props.className },
    'Hello from ',
    props.phrase,
    '!'
  );
};

StorybookExample.propTypes = {
  phrase: _react2.default.PropTypes.string.isRequired,
  className: _react2.default.PropTypes.string
};

StorybookExample.defaultProps = {
  phrase: 'a Stranger',
  className: 'StorybookExample--primary'
};

exports.default = StorybookExample;