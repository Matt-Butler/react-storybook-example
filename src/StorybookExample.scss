.StorybookExample--primary {
  background-color: #a77;
  color: #fff;
}

.StorybookExample--secondary {
  background-color: #e00;
  color: #fff;
}

.StorybookExample--negative {
  background-color: #0f0;
  color: #fff;
}
