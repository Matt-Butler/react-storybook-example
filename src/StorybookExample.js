import React from 'react';
import '../src/StorybookExample.scss';

const StorybookExample = props => <h1 className={props.className}>Hello from {props.phrase}!</h1>;

StorybookExample.propTypes = {
  phrase: React.PropTypes.string.isRequired,
  className: React.PropTypes.string,
};

StorybookExample.defaultProps = {
  phrase: 'a Stranger',
  className: 'StorybookExample--primary',
};

export default StorybookExample;
