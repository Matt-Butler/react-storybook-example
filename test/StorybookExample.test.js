import React from 'react';
import initStoryshots from 'storyshots';
import StorybookExample from '../src/StorybookExample.js';

// Run snapshot tests for react-storybook
initStoryshots();

it('should shallow my modal', () => {
    const wrapper = shallow(
        <StorybookExample phrase="Jest" />
    );
    expect(wrapper).toMatchSnapshot();
});

it('should mount my modal', () => {
    const wrapper = mount(
        <StorybookExample phrase="Jest" />
    );
    expect(wrapper).toMatchSnapshot();
});


it('should render my modal', () => {
    const wrapper = render(
        <StorybookExample phrase="Jest" />
    );
    expect(wrapper).toMatchSnapshot();
});
