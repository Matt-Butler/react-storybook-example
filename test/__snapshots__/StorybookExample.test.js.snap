exports[`Storyshots Modal Context Positive 1`] = `
<h1
  className="StorybookExample--negative">
  Hello from 
  a Stranger
  !
</h1>
`;

exports[`Storyshots Modal Context Primary 1`] = `
<h1
  className="StorybookExample--primary">
  Hello from 
  a Stranger
  !
</h1>
`;

exports[`Storyshots Modal Context Secondary 1`] = `
<h1
  className="StorybookExample--secondary">
  Hello from 
  a Stranger
  !
</h1>
`;

exports[`Storyshots StorybookExample Phrase With default props 1`] = `
<h1
  className="StorybookExample--primary">
  Hello from 
  a Stranger
  !
</h1>
`;

exports[`Storyshots StorybookExample Phrase With phrase Matt 1`] = `
<h1
  className="StorybookExample--primary">
  Hello from 
  Matt
  !
</h1>
`;

exports[`Storyshots StorybookExample Phrase With phrase the other side 1`] = `
<h1
  className="StorybookExample--primary">
  Hello from 
  the other side
  !
</h1>
`;

exports[`test should mount my modal 1`] = `
<StorybookExample
  className="StorybookExample--primary"
  phrase="Jest">
  <h1
    className="StorybookExample--primary">
    Hello from 
    Jest
    !
  </h1>
</StorybookExample>
`;

exports[`test should render my modal 1`] = `
<h1
  class="StorybookExample--primary">
  Hello from Jest!
</h1>
`;

exports[`test should shallow my modal 1`] = `
<h1
  className="StorybookExample--primary">
  Hello from 
  Jest
  !
</h1>
`;
